packages/core/src/App/Components/Layout/Footer/toggle-language-settings.tsx

```diff
@@ -10,9 +10,13 @@ import { TranslationFlag } from '@deriv/shared';
 
 const ToggleLanguageSettings = observer(({ showPopover }: { showPopover?: boolean }) => {
     const { common, ui } = useStore();
-    const { currentLang } = useTranslations();
+    const { currentLang, switchLanguage } = useTranslations();
     const { is_language_settings_modal_on, toggleLanguageSettingsModal } = ui;
-    const { is_language_changing } = common;
+    const { is_language_changing, current_language } = common;
+
+    React.useEffect(() => {
```

Comment:
This should be moved to `app.jsx` - since you would want the sync to happen globally

Reply:
Thank you for pointing it @yashim-deriv .  It is not needed as the value is already there in app.jsx